polygon-bridged-carbon/src/ToucanCarbonOffsets.ts

```diff
@@ -30,24 +30,28 @@ export function handleTransfer(event: Transfer): void {
     //carbonOffset.bridges.push(bridge.id)
   }
 
-  // Handle Retirements
-  if (event.params.to == Address.fromString('0x0000000000000000000000000000000000000000')) {
-    // This is a temporary solution - we are not tracking retirements related to HFC-23 Credits
-    // What needs to be done is to track Retire events instead of Transfer
-    if (carbonOffset.methodology != 'AM0001') {
-      let retire = loadOrCreateRetire(carbonOffset, transaction)
-      retire.value = toDecimal(event.params.value, 18)
-      retire.retiree = event.params.from.toHexString()
+  carbonOffset.currentSupply = toDecimal(offsetERC20.totalSupply(), 18)
+  carbonOffset.lastUpdate = transaction.timestamp
 
-      retire.save()
+  carbonOffset.save()
+}
 
-      carbonOffset.totalRetired = carbonOffset.totalRetired.plus(toDecimal(event.params.value, 18))
-      CarbonMetricUtils.updateCarbonTokenRetirements(new TCO2(), event.block.timestamp, event.params.value)
-    }
-  }
+export function handleRetired(event: Retired): void {
+  let transaction = loadOrCreateTransaction(event.transaction, event.block)
+  let offsetERC20 = ToucanCarbonOffsets.bind(event.address)
 
-  carbonOffset.currentSupply = toDecimal(offsetERC20.totalSupply(), 18)
-  carbonOffset.lastUpdate = transaction.timestamp
+  let carbonOffset = loadOrCreateCarbonOffset(transaction, event.address, 'Toucan', 'Verra')
+
+  if (carbonOffset.methodology != 'AM0001') {
```

Comment:
This is for excluding the HFC-23s, right?

We should double check with Drew & co about this, there is an open question as to whether the HFC-23s should be included in the retirement figures